sources/packages/backend/apps/test-db-seeding/src/db-seeding/02-institution/05-manage-designations.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { DesignationAgreement, InstitutionLocation } from "@sims/sims-db";
+import {
+  createFakeDesignationAgreement,
+  createMultipleFakeInstitutionLocations,
+} from "@sims/test-utils";
+import { Repository } from "typeorm";
+import {
+  DataSeed,
+  DataSeedMethod,
+  SeedPriorityOrder,
+} from "../../seed-executors";
+import { InstitutionHelperService } from "../../services";
+import { SIMS_COLL_F_LEGAL_SIGNING_USER } from "../constants";
+
+@Injectable()
+@DataSeed({ order: SeedPriorityOrder.Unknown })
+export class DesignationAgreementService {
+  constructor(
+    @InjectRepository(DesignationAgreement)
+    private readonly designationAgreementRepo: Repository<DesignationAgreement>,
```

Comment:
IMO this way of using repo looks a lot better than the base class way that we do in API.

Reply:
@dheepak-aot, it looks better to me too. I was willing to get rid of the base class that we use in the API for a long time. @ann-aot and I talked about starting the change in this library. I would suggest moving to this everywhere.